woocommerce/woocommerce-android
# Merge junit reports into one file

To speed up test collection time




### Description


#### Problem to address
Collecting JUnit results for unit tests takes significant part of the unit tests job (3m for Buildkite annotations, 4m for Buildkite test collector, internal source: paaHJt-6MN-p2 )

#### Possible solution
I've noticed that the problem is visible in unit tests, which produces a separate `.xml` report file per tests-suite, rather than in instrumented tests, where there's a single `.xml` report.

#### Why unit tests produces multiple `.xml` reports and instrumented tests produce single?
It seems it's the way AGP is organizing running instrumented tests. According to my research, there's no way to communicate to JUnit to produce a single report file, hence a simple merge logic proposed in this PR.

I think it's better to use this custom, small piece of logic than e.g. intrudcing a 3rd party npm package like https://www.npmjs.com/package/junit-report-merger

### Testing information

1. Go to a `trunk` commit that this PR is based on: 2a8a3d1
2. Check build results, `unit-tests` job
3. Tap on `View in test analytics` 
4. Check the `Test execution count` (it's 3744)
5. Verify this is the same value of collected test results as in this job from this PR.

### Future actions
I'd like to merge this PR as it is and see how impactful this change is. After asserting the impact and lack of problems, I'd like to move `merge_junit.sh` script to our CI toolkit, to make it available for both WC and WP/JP Android projects.
